extra/resources/HealthCPUIoWait

```diff
@@ -0,0 +1,177 @@
+. ${OCF_FUNCTIONS}
+: ${__OCF_ACTION=$1}
+
+#######################################################################
+
+meta_data() {
+        cat <<END
+<?xml version="1.0"?>
+<!DOCTYPE resource-agent SYSTEM "ra-api-1.dtd">
+<resource-agent name="CPUIdle" version="1.0">
+<version>1.0</version>
+
+<longdesc lang="en">
+Systhem health agent that measures the CPU iowait via top and updates the #health-cpuiowait attribute.
+</longdesc>
+<shortdesc lang="en">System health CPU iowait measurement</shortdesc>
+
+<parameters>
+<parameter name="state" unique="1">
+<longdesc lang="en">
+Location to store the resource state in.
+</longdesc>
+<shortdesc lang="en">State file</shortdesc>
+<content type="string" default="${HA_VARRUN%%/}/health-cpuiowait-${OCF_RESOURCE_INSTANCE}.state" />
+</parameter>
+
+<parameter name="yellow_limit" unique="1">
+<longdesc lang="en">
+Upper limit of iowait percentage to switch the health attribute to yellow. I.e.
+the #health-cpuiowait will go yellow if the %iowait of the CPU gets higher than 10%.
+</longdesc>
+<shortdesc lang="en">Upper limit for yellow health attribute</shortdesc>
+<content type="string" default="10"/>
+</parameter>
+
+<parameter name="red_limit" unique="1">
+<longdesc lang="en">
+Upper limit of iowait percentage to switch the health attribute to red. I.e.
+the #health-cpuiowait will go red if the %iowait of the CPU get higher than 15%.
+</longdesc>
+<shortdesc lang="en">Upper limit for red health attribute</shortdesc>
+<content type="string" default="15"/>
+</parameter>
+
+</parameters>
+
+<actions>
+<action name="start"        timeout="10s" />
+<action name="stop"         timeout="10s" />
+<action name="monitor"      timeout="10s" interval="10s" start-delay="0s" />
+<action name="meta-data"    timeout="5s" />
+<action name="validate-all" timeout="10s" />
+</actions>
+</resource-agent>
+END
+}
+
+#######################################################################
+
+agent_usage() {
+        cat <<END
+usage: $0 {start|stop|monitor|validate-all|meta-data}
+
+Expects to have a fully populated OCF RA-compliant environment set.
+END
+}
+
+agent_start() {
+    agent_monitor
+    if [ $? =  $OCF_SUCCESS ]; then
+      return $OCF_SUCCESS
+    fi
+    touch "${OCF_RESKEY_state}"
+}
+
+agent_stop() {
+    agent_monitor
+    if [ $? =  $OCF_SUCCESS ]; then
+        rm ${OCF_RESKEY_state}
+    fi
+    return $OCF_SUCCESS
+}
+
+agent_monitor() {
+        # Monitor _MUST!_ differentiate correctly between running
+        # (SUCCESS), failed (ERROR) or _cleanly_ stopped (NOT RUNNING).
+        # That is THREE states, not just yes/no.
+        if [ -f ${OCF_RESKEY_state} ]; then
+          WAIT=`top -b -n2 | grep Cpu | tail -1 | awk -F",|.[0-9][ %]wa" '{ print $5 }'`
+          # echo "System iowait: " $WAIT
+          # echo $OCF_RESKEY_yellow_limit
+          if [ $WAIT -gt ${OCF_RESKEY_red_limit} ] ; then
+            # echo "System state RED!"
+            attrd_updater -n "#health-cpuiowait" -U "red" -d "30s"
```

Comment:
30s is probably too long; it would make any monitor interval lower than that ineffective. HealthSMART's 5s is probably a good value (I need to fix it in HealthCPU ...)